proteus/NumericalSolution.py

```diff
@@ -935,6 +979,265 @@ def calculateSolution(self,runName):
+                            assert(lmOld.timeIntegration.t == lm.timeIntegration.t)
+                            assert(lmOld.timeIntegration.dt == lm.timeIntegration.dt)
+                            #lm.coefficients.evaluate(self.t_stepSequence,lm.q)
+                            #lm.coefficients.evaluate(self.t_stepSequence,lm.ebqe)
+                            #lm.timeIntegration.calculateElementCoefficients(lm.q)
+                        assert(m.stepController.dt_model == mOld.stepController.dt_model)
+                        assert(m.stepController.t_model == mOld.stepController.t_model)
+                        assert(m.stepController.t_model_last == mOld.stepController.t_model_last)
+                        logEvent("Initializing time history for model step controller")
+                        m.stepController.initializeTimeHistory()
+                    p0.domain.initFlag=True #For next step to take initial conditions from solution, only used on restarts
+                    self.systemStepController.modelList = self.modelList
+                    self.systemStepController.exitModelStep = {}
+                    self.systemStepController.controllerList = []
+                    for model in self.modelList:
+                        self.systemStepController.exitModelStep[model] = False
+                        if model.levelModelList[-1].timeIntegration.isAdaptive:
+                            self.systemStepController.controllerList.append(model)
+                            self.systemStepController.maxFailures = model.stepController.maxSolverFailures
+                    self.systemStepController.choose_dt_system()
+                    for m,ptmp,mOld in zip(self.modelList, self.pList, modelListOld):
+                        for lm, lu, lr, lmOld in zip(m.levelModelList, m.uList, m.rList, mOld.levelModelList):
+                            assert(lmOld.timeIntegration.tLast == lm.timeIntegration.tLast)
+                            assert(lmOld.timeIntegration.t == lm.timeIntegration.t)
+                            assert(lmOld.timeIntegration.dt == lm.timeIntegration.dt)
+                        assert(m.stepController.dt_model == mOld.stepController.dt_model)
+                        assert(m.stepController.t_model == mOld.stepController.t_model)
+                        assert(m.stepController.t_model_last == mOld.stepController.t_model_last)
+                    if self.archiveFlag == ArchiveFlags.EVERY_SEQUENCE_STEP:
+                        #hack for archiving initial solution on adapted mesh
+                        self.tCount+=1
+                        for index,model in enumerate(self.modelList):
+                            self.archiveSolution(
+                                model,
+                                index,
+                                self.systemStepController.t_system_last+1.0e-6)
+                  ##chitak end Adapt
```

Comment:
We should probably put this adaption block into a member function and call it here.